rust/xaynet-server/src/storage/api.rs

```diff
@@ -0,0 +1,240 @@
+    ///
+    /// # Behavior
+    ///
+    /// - If the local seed dict has been successfully added, return `Result::Ok(LocalSeedDictAdd(Result::Ok(())))`.
+    /// - If the local seed dict could not be added due to a PET protocol error, return
+    ///   the corresponding `Result::Ok(LocalSeedDictAdd(Result::Err(LocalSeedDictAddError)))`.
+    async fn add_local_seed_dict(
+        &mut self,
+        update_pk: &UpdateParticipantPublicKey,
+        local_seed_dict: &LocalSeedDict,
+    ) -> StorageResult<LocalSeedDictAdd>;
+
+    /// Gets the [`SeedDict`].
+    ///
+    /// # Behavior
+    ///
+    /// - If the seed dict does not exist, return `Result::Ok(Option::None)`.
+    /// - If the seed dict exists, return `Result::Ok(Option::Some(SeedDict))`.
+    async fn get_seed_dict(&mut self) -> StorageResult<Option<SeedDict>>;
+
+    /// Increments the mask score with the given [`MaskObject`].
+    /// The score of the given mask is incremented by `1`.
+    ///
+    /// # Behavior
+    ///
+    /// - If the mask score has been successfully incremented, return `Result::Ok(Vec<(MaskObject, u64)>)`.
+    /// - If the mask score could not be incremented due to a PET protocol error, return
+    ///   the corresponding `Result::Ok(MaskScoreIncr(Result::Err(MaskScoreIncrError)))`.
+    async fn incr_mask_score(
+        &mut self,
+        pk: &SumParticipantPublicKey,
+        mask: &MaskObject,
+    ) -> StorageResult<MaskScoreIncr>;
+
+    /// Gets the two masks with the highest score.
+    ///
+    /// # Behavior
+    ///
+    /// - If no masks exist, return `Result::Ok(Option::None)`.
+    /// - If only one mask exists, return this mask
+    ///   `Result::Ok(Option::Some(Vec<(MaskObject, u64)>))`.
+    /// - If two masks exist with the same score, return both
+    ///   `Result::Ok(Option::Some(Vec<(MaskObject, u64)>))`.
+    /// - If two masks exist with the different score, return
+    ///   both in descending order `Result::Ok(Option::Some(Vec<(MaskObject, u64)>))`.
+    async fn get_best_masks(&mut self) -> StorageResult<Option<Vec<(MaskObject, u64)>>>;
+
+    /// Gets the number of unique masks.
+    async fn get_number_of_unique_masks(&mut self) -> StorageResult<u64>;
+
+    /// Deletes all coordinator data. This includes the coordinator
+    /// state as well as the [`SumDict`], [`SeedDict`] and `mask` dictionary.
+    async fn delete_coordinator_data(&mut self) -> StorageResult<()>;
+
+    /// Deletes the [`SumDict`], [`SeedDict`] and `mask` dictionary.
+    async fn delete_dicts(&mut self) -> StorageResult<()>;
+
+    /// Sets the latest global model id.
+    ///
+    /// # Behavior
+    ///
+    /// - If no global model id has been set yet, set and return the new id `Result::Ok(String))`.
+    /// - If the global model id already exists, override and return the new id `Result::Ok(String))`.
+    async fn set_latest_global_model_id(&mut self, id: &str) -> StorageResult<()>;
```

Comment:
description doesn't match the return type, either

```suggestion
    /// - If no global model id has been set yet, set and return the new id `StorageResult::Ok(String)`.
    /// - If the global model id already exists, override and return the new id `StorageResult::Ok(String)`.
    async fn set_latest_global_model_id(&mut self, id: &str) -> StorageResult<String>;
```

or

```suggestion
    /// - If no global model id has been set yet, set the new id `StorageResult::Ok(())`.
    /// - If the global model id already exists, override with the new id `StorageResult::Ok(())`.
    async fn set_latest_global_model_id(&mut self, id: &str) -> StorageResult<()>;
```